Src/WindowsServer/WindowsServer.Shared/Implementation/AppServiceEnvVarMonitor.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Utility to monitor the value of environment variables which may change 
+    /// during the run of an application. Checks the environment variables 
+    /// intermittently.
+    /// </summary>
+    internal static class AppServiceEnvVarMonitor
+    {
+        // Environment variables tracked by this monitor. (internal to allow tests to modify them)
+        internal static readonly Dictionary<string, string> CheckedValues = new Dictionary<string, string>()
+        {
+            { "WEBSITE_SITE_NAME", string.Empty },
+            { "WEBSITE_HOME_STAMPNAME", string.Empty },
+            { "WEBSITE_HOSTNAME", string.Empty }
```

Comment:
In comments for environment variable extraction in [this PR](https://github.com/Azure/azure-webjobs-sdk/pull/1382/files), it's noted that WEBSITE_HOSTNAME is not a reliable source:

> //...refactored so that it did not use WEBSITE_HOSTNAME, which is determined to be unreliable for functions during slot swaps. 

Do we want to rely on that field, or we may use some tricks, e.g. from the same PR to build something more reliable from what we have.

Reply:
I say we should resort to these tactics if we have an issue or have proof that App Services are affected by the same failing that Azure Functions are. I have put an issue together to verify that we can detect slot swap using only WEBSITE_HOSTNAME, and to update to using the WEBSITE_SITE_NAME + WEBSITE_SLOT_NAME if the need arises. Note that I've included the WEBSITE_SLOT_NAME in the list of environment variables that the monitor watches for now.